packages/utils/telemetry-utils/src/test/sampledTelemetryHelper.spec.ts

```diff
@@ -233,6 +234,95 @@ describe("SampledTelemetryHelper", () => {
 		helper.dispose();
 		assert.strictEqual(logger.events.length, 1);
 	});
+
+	it("Correctly returns computed duration for custom data", () => {
+		const sampling = 10;
+		const helper = new SampledTelemetryHelper(
+			{ eventName: "testEvent" },
+			logger,
+			sampling,
+			true /* includeAggregateMetrics */,
+		);
+
+		const clock = sinon.useFakeTimers();
+		const startingPoint = 50; // Arbitrary starting point.
+		let totalDuration = 0;
+		let maxDuration = Number.MIN_VALUE;
+		let minDuration = Number.MAX_VALUE;
+
+		for (let i = 0; i < sampling; i++) {
+			helper.measure(() => {
+				const currentIterationDuration = startingPoint + i;
+
+				clock.tick(currentIterationDuration);
+				totalDuration += currentIterationDuration;
+				maxDuration = Math.max(maxDuration, currentIterationDuration);
+				minDuration = Math.min(minDuration, currentIterationDuration);
+			});
+		}
+
+		clock.restore();
+
+		assert.strictEqual(logger.events.length, 1);
+		assert.strictEqual(logger.events[0].totalDuration, totalDuration);
+		assert.strictEqual(logger.events[0].averageDuration, totalDuration / sampling);
+		assert.strictEqual(logger.events[0].maxDuration, maxDuration);
+		assert.strictEqual(logger.events[0].minDuration, minDuration);
+	});
+});
+
+/**
+ * @remarks Initialized in advance to extract its keys for type checking.
+ * Arbitrary properties that can be logged with the telemetry event.
+ */
+interface TestTelemetryProperties {
+	propertyOne: number;
+	propertyTwo: number;
+	propertyThree: number;
+}
+
+describe("SampledTelemetryHelper with Custom Data", () => {
```

Comment:
I think no need for the extra suite, just put the new test inside the suite above.